hikvision_tftpd.py

```diff
@@ -130,17 +143,20 @@ def _tftp_read(self):
         else:
             print '%s: received unexpected tftp bytes %r from %s:%d' % (
                 now, pkt.encode('hex'), addr[0], addr[1])
+
     def _tftp_options_ack(self, addr):
-        pkt = (struct.pack('>H', self._TFTP_OPCODE_OACK) + 'blksize' + '\x00' + str(self.BLOCK_SIZE) + '\x00')
+        pkt = (struct.pack('>H', self._TFTP_OPCODE_OACK) + 'blksize\x00' + str(self._block_size) + '\x00')
         self._tftp_sock.sendto(pkt, addr)
 
     def _tftp_maybe_send(self, prev_block, addr):
         block = prev_block + 1
-        start_byte = prev_block * self.BLOCK_SIZE
+        start_byte = prev_block * self._block_size
         if start_byte > len(self._file_contents):
             print '%s: done!' % time.strftime(_TIME_FMT)
+            if self._block_size != _DEFAULT_BLOCK_SIZE:
+                self.set_default_block_size()
```

Comment:
could this also be `self._set_block_size(_DEFAULT_BLOCK_SIZE)`, so we can get rid of the duplicate `set_default_block_size`?

Reply:
The issue with calling self._set_block_size() was the max blocks check would cause the server to raise and error when setting the default block size back to 512 if the file your serving is >32 MB.

```if self._total_blocks > 65535:
            raise Error('File is too big to serve with %d-byte blocks.'
                        % self._block_size)```

I am open to suggestions here.

Reply:
Oh, good point. Yeah, keeping that method separate sounds fine, or maybe the raise should be in a separate method which is called in the two places we actually send the first block.